### PR TITLE
Fix unused functions that produce bugs (related to #529)

### PR DESCRIPTION
**Pull request by KitHat**
These two variables seem to be unused by the code and produce bugs like #529 
Can we omit them?

### CODE REVIEW

**Comment by johnynek** [comment]
Thanks for the PR.

There are a few things wrong here: 1. We shouldn't use strings to get these classes. We should just make a list 23 elements long and use `List(classOf[Function0], classOf[Function1], ...`

Secondly, we shouldn't have a function like this if we aren't using it in general, BUT I think removing it will break binary compatibility (mima should fail in the CI).

I guess the easier thing it to fix the code not to use forName and reflection here, that way we fix your issue but also don't break binary compatibility.

How does that sound?

**Comment by KitHat** [comment]
@johnynek sounds good, I will fix it.

**Comment by johnynek** [comment]
we shouldn't need to comment out the protobuf test.

Can you merge master, and not comment that out and make sure things still pass?

File: chill-scala/src/main/scala/com/twitter/chill/KryoBase.scala [hunk 1]
```diff
@@ -48,9 +47,29 @@ class KryoBase(classResolver: ClassResolver, referenceResolver: ReferenceResolve
   protected var strategy: Option[InstantiatorStrategy] = None
 
   val functions: Iterable[Class[_]] =
-    (0 to 22).map { idx =>
-      Class.forName("scala.Function" + idx.toString, true, Thread.currentThread().getContextClassLoader())
-    }
+    List(
```
Comment (johnynek) [inline chill-scala/src/main/scala/com/twitter/chill/KryoBase.scala]: thanks for fixing this.... some of this code is pretty old and really not optimally written but also used in a variety of big systems (many versions of spark, scalding, and a few other systems) so being committed to binary compatibility really helps avoid nasty diamond dependency problems.

**Review by johnynek — CHANGES_REQUESTED** [review]

**Comment by KitHat** [comment]
Master merged, launched tests with `sbt test` -- they have passed.
Uncommented the Protobuf test as well -- it was commented out by mistake

**Review by johnynek — APPROVED** [review]
Thanks so much!

File: chill-scala/src/main/scala/com/twitter/chill/KryoBase.scala [hunk 2]
```diff
@@ -48,9 +47,29 @@ class KryoBase(classResolver: ClassResolver, referenceResolver: ReferenceResolve
...
+      classOf[Function9[_,_,_,_,_,_,_,_,_,_]],
+      classOf[Function10[_,_,_,_,_,_,_,_,_,_,_]],
+      classOf[Function11[_,_,_,_,_,_,_,_,_,_,_,_]],
+      classOf[Function12[_,_,_,_,_,_,_,_,_,_,_,_,_]],
+      classOf[Function13[_,_,_,_,_,_,_,_,_,_,_,_,_,_]],
+      classOf[Function14[_,_,_,_,_,_,_,_,_,_,_,_,_,_,_]],
+      classOf[Function15[_,_,_,_,_,_,_,_,_,_,_,_,_,_,_,_]],
+      classOf[Function16[_,_,_,_,_,_,_,_,_,_,_,_,_,_,_,_,_]],
+      classOf[Function17[_,_,_,_,_,_,_,_,_,_,_,_,_,_,_,_,_,_]],
+      classOf[Function18[_,_,_,_,_,_,_,_,_,_,_,_,_,_,_,_,_,_,_]],
```
Comment (regadas) [inline chill-scala/src/main/scala/com/twitter/chill/KryoBase.scala]: Just noticed this PR (sorry for poking); Here we jumped `Function19`

Comment (KitHat) [reply chill-scala/src/main/scala/com/twitter/chill/KryoBase.scala]: You have an eagle eye!
Fixed this along with scalafmt

**Comment by KitHat** [comment]
> Thanks so much!

I have fixed the scalafmt issue as well

**Pull request merged**